src/logic/calculate.js

```diff
@@ -0,0 +1,18 @@
+import operate from './operate';
+
+const calculate = ({
+  result, nextVal, operation, buttonVal,
+}) => {
+  let res = result;
+  if (buttonVal === 'Ac') {
+    res = 0;
+    return result;
+  }
+  if (buttonVal === '+/-') {
+    res *= -1;
+    return result;
+  }
+  return operate(res, nextVal, operation);
+};
+
+export default calculate;
```

Comment:
- Also we are missing something very important as stated in the Pathwright to this Milestone](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896094/)
![image](https://user-images.githubusercontent.com/46115725/73102587-ca2b6400-3eb7-11ea-8c1a-7611b61564e6.png)

This means that from our second argument, the buttonVal, which is going to take into account the name of the button that was clicked by the user, the properties of the data object should behave accordingly e.g. if `if(buttonVal==='AC'){total: null, next: null, operation: null}` since the function of that particular button is to clear the calculator of all ongoing proceses. So we need to try and cover every possible interaction that might come from recieving the buttonVal argument, meaning, if the user presses a button that is a number, and then an operation, and then another number, what is the interaction that this should have with the data object.